### PR TITLE
feat(emulator): take in options as an optional argument

### PR DESCRIPTION
**Pull request by felixmr1**
Since we are not running this in real time in the simulator, we are constantly timing out. Therefore we want a way to disable it.

### CODE REVIEW

File: client_test.go [hunk 1]
```diff
@@ -129,7 +129,21 @@ func TestClient_GetOutputConfiguration(t *testing.T) {
 		},
 	}
 
-	reqOutputConfigurationAck := []byte{0xfa, 0xff, 0xc1, 0x8, 0x50, 0x40, 0x0, 0x64, 0x20, 0x13, 0x0, 0xc8, 0x49}
```
Comment (felixmr1) [inline client_test.go:132]: My LSP made some automatic changes here, due to line length. Can revert if you guys want to!

File: xsensemulator/udpserialport.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
+package xsensemulator
```
Comment (felixmr1) [inline xsensemulator/udpserialport.go:1]: moved the UDPSerialPort to its own file.

File: xsensemulator/udpserialport.go [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+// UDPSerialPortOption configures an UDPSerialPort.
+type UDPSerialPortOption func(*udpSerialPortOptions)
+
+// WithTransmitInterface configures the interface to transmit on.
+func WithTimeout(timeout time.Duration) UDPSerialPortOption {
+	return func(opt *udpSerialPortOptions) {
+		opt.timeout = timeout
+	}
+}
```
Comment (felixmr1) [inline xsensemulator/udpserialport.go:91]: Added the opts pattern here.

File: client_test.go (hunk 1 above)
Comment (gust1n) [reply client_test.go:132]: Could you make 2 separate commits so its easier to follow what the actual changes are?

File: xsensemulator/udpserialport.go [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+func (t UDPSerialPort) Close() error {
+	return t.OriginConn.Close()
+}
+
+type udpSerialPortOptions struct {
+	// Timeout for setting read/write deadlines
+	timeout time.Duration
+}
+
+// defaultTransmitterOptions returns transmitter options with sensible default values.
```
Comment (gust1n) [inline xsensemulator/udpserialport.go:78]: wrong name in comment

**Review by gust1n — APPROVED** [review]
LGTM! Would have been easier to follow changes if split in separate commits, like one for lint fixes, one for moving the struct to a separate file and then one with the actual changes, so feel free to arrange like that before merging.

**Comment by felixmr1** [comment]
> LGTM! Would have been easier to follow changes if split in separate commits, like one for lint fixes, one for moving the struct to a separate file and then one with the actual changes, so feel free to arrange like that before merging.

Agreed, I removed the linting since that was non intentional. I kept it as one commit, though. Was to much of an hassel to make partial file commits, I will do it like that next time instead :)

**Pull request merged**